server/sonar-server/src/main/java/org/sonar/server/user/ServerUserSession.java

```diff
@@ -110,8 +109,17 @@ public Integer getUserId() {
   }
 
   @Override
+  public Collection<GroupDto> getGroups() {
+    return groups.get();
+  }
+
+  @Override
   public Set<String> getUserGroups() {
-    return userGroups;
+    Collection<GroupDto> groups = getGroups();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "groups" which hides the field declared at line 59. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)